mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -766,9 +776,20 @@ LogicalResult ThreadwiseReadIntoRewritePattern::matchAndRewrite(
   SmallVector<int64_t, 3> bounds{1, numValues};
   SmallVector<int64_t, 3> strides{1, vectorLen};
   if (srcAddrSpace == gpu::AddressSpace::Global) {
-    readStartCoords.insert(readStartCoords.begin(), bid);
-    bounds.insert(bounds.begin(), 1);
-    strides.insert(strides.begin(), 1);
+    SmallVector<Value> blockCoords;
+    if (inputShape.size() == 3) {
+      blockCoords = SmallVector<Value>{bid};
+    } else {
+      int64_t numCU = 110;
```

Comment:
We dont need to this after https://github.com/ROCmSoftwarePlatform/rocMLIR/pull/1134.
i.e. we dont need this logic recalculated at this level in the lowering.

Please have a look at the PR and I think could simply pass in Values : g_block, m_block and n_block when threadwise_read / threadwise_write ops are created.